hibernate-core/src/main/java/org/hibernate/metamodel/mapping/internal/ToOneAttributeMapping.java

```diff
@@ -1191,8 +1191,8 @@ public EntityFetch generateFetch(
 				&& parentNavigablePath.equals( fetchParent.getNavigablePath().getRealParent() );
 
 
-		if ( (hasNotFoundAction()
-				|| ( fetchTiming == FetchTiming.IMMEDIATE && selected )) ) {
+		if ( ( ( hasNotFoundAction() && doesNotExistATableGroupJoin( parentTableGroup, fetchablePath ) )
```

Comment:
Would you mind explaining why we need `doesNotExistATableGroupJoin( parentTableGroup, fetchablePath )`? Maybe even as code comment?

Reply:
Unfortunately it seems not so easy :( I tried but stumbled in a lot of issues.
I tried different ways but without any success

Reply:
Let me know what do you think about the comment I added